includes/currency/class-money-formatter.php

```diff
@@ -113,6 +113,37 @@ public function format_for_display( $decimals = true ) {
 		return $this;
 	}
 
+	/**
+	 * Formats the amount for typed data returns.
+	 * Does not apply the currency code and returns a foat instead of a string.
+	 *
+	 * @since 3.0
+	 * @return Money_Formatter
+	 */
+	public function format_for_typed( $decimals = true ) {
+		$amount = $this->unformat();
+
+		if ( empty( $amount ) ) {
+			$amount = 0;
+		}
+
+		/**
+		 * Filter number of decimals to use for formatted amount
+		 *
+		 * @since unknown
+		 *
+		 * @param int        $number        Default 2. Number of decimals.
+		 * @param int|string $amount        Amount being formatted.
+		 * @param string     $currency_code Currency code being formatted.
+		 */
+		$decimals = apply_filters( 'edd_format_amount_decimals', $decimals ? $this->currency->number_decimals : 0, $amount, $this->currency->code );
+
+		// Format amount using decimals and separators (also rounds up or down).
+		$formatted = number_format( (float) $amount, $decimals, $this->currency->decimal_separator, $this->currency->thousands_separator );
```

Comment:
I think this is actually not correct--shouldn't this be assigned to `$this->amount` instead of just a local variable? I'm thinking the current assignment actually means that the result is never used.

If I change it locally to `$this->amount` then the browser results are actually what I expect to see, although they are strings, not floats. But I think right now this method isn't actually modifying the end result.

Reply:
@robincornett Updated, unit tests still pass, have a look at the change in https://github.com/awesomemotive/easy-digital-downloads/pull/9362/commits/e3ec422b9bcf853764fd5c3e9a71220d40d2d2ef